app/src/main/java/com/google/firebase/ml/md/java/camera/CameraSourcePreview.java

```diff
@@ -106,7 +106,7 @@ protected void onLayout(boolean changed, int left, int top, int right, int botto
     int childHeight = (int) (childWidth / previewSizeRatio);
     if (childHeight <= layoutHeight) {
       for (int i = 0; i < getChildCount(); ++i) {
-        getChildAt(i).layout(0, 0, childWidth, childHeight);
+        getChildAt(i).layout(0, 0, childWidth, getHeight());
```

Comment:
IIRC matching container's height forcibly will cause the camera frame distortion if the aspect ratio of camera preview is different than the container. For example, camera ratio is 16 : 9 and container ratio is 17 : 9, then this change will stretch the camera frame to have ratio 17 : 9.